src/frequenz/sdk/timeseries/logical_meter/_formula_engine.py

```diff
@@ -8,10 +8,11 @@
 import asyncio
```

Comment:
Would it be ok to separate this file (_formula_engine.py) into 3 files?
* _base_formula_engine.py
* _formula_engine.py
* _3Phase_formula_engine.py

Generics can stay. I am telling only about separating this 1 file .

Reply:
Unfortunately,  to do this,  we'll have to do some sacrifices to type safety,  or else, there will be cyclic imports.  

Specifically,  the generic types `_GenericEngine`, etc will have to be unbounded,  because otherwise the base classes file will have to import each engine implementation separately.

So we'll have `_GenericEngine = TypeVar("_GenericEngine")` instead of `= TypeVar("_GenericEngine", FormulaEngine, FormulaEngine3Phase)`
and `ReceiverType: Type[Any]` instead of `: Type[FormulaReceiver | FormulaReceiver3Phase]`.

I think that's tighter constraints and better type checks are more valuable than smaller files.  But I guess it is a tough decision.